libyul/backends/evm/ControlFlow.h

```diff
@@ -39,14 +49,19 @@ struct ControlFlow
 		return nullptr;
 	}
 
-	std::string toDot() const
+	std::string toDot(ControlFlowLiveness const* _liveness=nullptr) const
 	{
 		std::ostringstream output;
 		output << "digraph SSACFG {\nnodesep=0.7;\ngraph[fontname=\"DejaVu Sans\"]\nnode[shape=box,fontname=\"DejaVu Sans\"];\n\n";
-		output << mainGraph->toDot(false);
-		size_t index = 1;
-		for (auto const& [function, functionGraph]: functionGraphMapping)
-			output << functionGraph->toDot(false, index++);
+		output << mainGraph->toDot(false, std::nullopt, _liveness ? _liveness->mainLiveness.get() : nullptr);
+
+		for (size_t index=0; index < functionGraphs.size(); ++index)
+			output << functionGraphs[index]->toDot(
+				false,
+				index+1,
+				_liveness ? _liveness->functionLiveness[index].get() : nullptr
```

Comment:
I'm wondering if it is guaranteed that `_liveness->functionLiveness.size() == functionGraphs.size()`. Maybe an assert before the for loop would be nice.
```c++
		yulAssert(_liveness->functionLiveness.size() == functionGraphs.size());
```

But also, I feel that we could do this differently, see my comment https://github.com/ethereum/solidity/pull/15439/files#r1767101120

Reply:
If you really want to assert that the liveness belongs to the control flow, we should just save a reference to the control flow in the liveness :) this assert gives an indication that they might belong together but nothing more